en/developer-docs/docs/develop-components/connect-to-remote-dependencies-while-developing.md

```diff
@@ -0,0 +1,121 @@
+To streamline local development and verification of Choreo components that depend on [Choreo connections](../../choreo-concepts/connections/), you can bridge your local environment with the remote/deployed project. This is facilitated by the Choreo CLI and VS Code extension, which create a secure connected environment enabling real-time testing and validation.
+
+This connected environment acts as a temporary link to your Choreo deployment. It provides your local component with the necessary connection configurations injected as environment variables. Furthermore, any outgoing HTTP requests from this local environment are securely routed to your deployed Choreo environment. This enables you to:
+
+- **Securely Access Dependencies**: Local components automatically get connection details for Choreo services, third-party services, and databases.
+- **Test Integrations Early**: Verify your local component's interaction with Choreo services before pushing code, using the same environment configuration.
+- **Consistent Testing Across Environments**: Ensure reliable and secure test results in both local and CI environments.
+
+## Using CLI
+
+The Choreo CLI allows you to create a local subshell environment that is bridged to your deployed Choreo environment.
+
+### Prerequisites
+
+1. Install the Choreo CLI. See the [setup guide](../../choreo-cli/get-started-with-the-choreo-cli/) for instructions.
+2. Login to Choreo via CLI
+
+    ``` sh
+    choreo login
+    ```
+
+!!! tip
+    For testing in CI environments, login to Choreo CLI using a [personal access token](../../choreo-cli/manage-authentication-with-personal-access-tokens/).    
+
+### Connect to a project
+
+1. The following command will create a subshell that will be connected to your remote project environment.
+
+    ``` sh
+    choreo connect --project default-project
+    ```
+
+2. Execute the command to run your component locally within the Choreo subshell.
+
+### Advanced examples
+
+- Connecting with the context of a specific component
+
+    ``` sh
+    choreo connect --project default-project --component my-component
+    ```
+
+- By default, you'll connect to the development environment. Use the command below to connect to other non-critical environments.
+
+    ``` sh
+    choreo connect --project default-project --env test-env
+    ```
+
+- By default, choreo connect injects all connection configurations. To use locally running dependencies instead, and skip injecting specific configurations, use the following command.
+
+    ``` sh
+    choreo connect --project default-project --skip-connection test-conn1 --skip-connection test-conn2 
+    ```
+
+- Combine the choreo connect command with your application's start command to create a subshell and launch your app within it.
+
+    ``` sh
+    choreo connect --project default-project -- go run main,go
```

Comment:
Isn't this the primary usecase, if so this should come on top

Reply:
I added `choreo connect --project default-project` as the default one as its more easier to understand